bug(TikTok): `Show seekbar` fails to apply on Latest TikTok Versions

### Type

Error while patching

### Bug description

Show Seekbar patch doesn't work on latest versions of TikTok
Tested version:
30.6.xx
30.7.xx

### Steps to reproduce

- Download ReVanced CLI, Patches, Integrations
- Try to patch TikTok or TikTok Asia with `show seekbar` patch include
- It'll fail to apply the patch

### Relevant log output

```shell
SEVERE: Show seekbar failed:
app.revanced.patcher.patch.PatchResultError: class org.jf.dexlib2.builder.instru
ction.BuilderInstruction35c cannot be cast to class org.jf.dexlib2.iface.instruc
tion.formats.Instruction22t (org.jf.dexlib2.builder.instruction.BuilderInstructi
on35c and org.jf.dexlib2.iface.instruction.formats.Instruction22t are in unnamed
 module of loader 'app')
        at app.revanced.patcher.Patcher.executePatches$executePatch(Patcher.kt:3
37)
        at app.revanced.patcher.Patcher.access$executePatches$executePatch(Patch
er.kt:38)
        at app.revanced.patcher.Patcher$executePatches$1.invokeSuspend(Patcher.k
t:358)
        at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(Contin
uationImpl.kt:33)
        at kotlin.sequences.SequenceBuilderIterator.hasNext(SequenceBuilder.kt:1
29)
        at app.revanced.utils.patcher.PatcherKt.applyPatchesVerbose(Patcher.kt:8
3)
        at app.revanced.cli.patcher.Patcher.start$revanced_cli(Patcher.kt:19)
        at app.revanced.cli.command.MainCommand.run(MainCommand.kt:173)
        at picocli.CommandLine.executeUserObject(CommandLine.java:2026)
        at picocli.CommandLine.access$1500(CommandLine.java:148)
        at picocli.CommandLine$RunLast.executeUserObjectOfLastSubcommandWithSame
Parent(CommandLine.java:2461)
        at picocli.CommandLine$RunLast.handle(CommandLine.java:2453)
        at picocli.CommandLine$RunLast.handle(CommandLine.java:2415)
        at picocli.CommandLine$AbstractParseResultHandler.execute(CommandLine.ja
va:2273)
        at picocli.CommandLine$RunLast.execute(CommandLine.java:2417)
        at picocli.CommandLine.execute(CommandLine.java:2170)
        at app.revanced.cli.main.MainKt.main(Main.kt:7)
Caused by: java.lang.ClassCastException: class org.jf.dexlib2.builder.instructio
n.BuilderInstruction35c cannot be cast to class org.jf.dexlib2.iface.instruction
.formats.Instruction22t (org.jf.dexlib2.builder.instruction.BuilderInstruction35
c and org.jf.dexlib2.iface.instruction.formats.Instruction22t are in unnamed mod
ule of loader 'app')
        at app.revanced.patches.tiktok.interaction.seekbar.patch.ShowSeekbarPatc
h.execute(ShowSeekbarPatch.kt:28)
        at app.revanced.patches.tiktok.interaction.seekbar.patch.ShowSeekbarPatc
h.execute(ShowSeekbarPatch.kt:17)
        at app.revanced.patcher.Patcher.executePatches$executePatch(Patcher.kt:3
33)
        ... 16 more

```

### Acknowledgements

- [X] I have searched the existing issues and this is a new and no duplicate or related to another open issue.
- [X] I have written a short but informative title.
- [X] I filled out all of the requested information in this issue properly.

--- a/src/main/kotlin/app/revanced/patches/tiktok/interaction/seekbar/fingerprints/ShouldShowSeekBarFingerprint.kt
+++ b/src/main/kotlin/app/revanced/patches/tiktok/interaction/seekbar/fingerprints/ShouldShowSeekBarFingerprint.kt
@@ -1,0 +1,9 @@
+package app.revanced.patches.tiktok.interaction.seekbar.fingerprints
+
+import app.revanced.patcher.fingerprint.method.impl.MethodFingerprint
+
+object ShouldShowSeekBarFingerprint : MethodFingerprint(
+    strings = listOf(
+        "can not show seekbar, state: 1, not in resume"
+    ),
+)

--- a/src/main/kotlin/app/revanced/patches/tiktok/interaction/seekbar/patch/ShowSeekbarPatch.kt
+++ b/src/main/kotlin/app/revanced/patches/tiktok/interaction/seekbar/patch/ShowSeekbarPatch.kt
@@ -5,14 +5,13 @@ import app.revanced.patcher.annotation.Description
 import app.revanced.patcher.annotation.Name
 import app.revanced.patcher.data.BytecodeContext
 import app.revanced.patcher.extensions.InstructionExtensions.addInstructions
-import app.revanced.patcher.extensions.InstructionExtensions.getInstruction
 import app.revanced.patcher.patch.BytecodePatch
 import app.revanced.patcher.patch.PatchResult
 import app.revanced.patcher.patch.PatchResultSuccess
 import app.revanced.patcher.patch.annotations.Patch
 import app.revanced.patches.tiktok.interaction.seekbar.annotations.ShowSeekbarCompatibility
 import app.revanced.patches.tiktok.interaction.seekbar.fingerprints.SetSeekBarShowTypeFingerprint
-import org.jf.dexlib2.iface.instruction.formats.Instruction22t
+import app.revanced.patches.tiktok.interaction.seekbar.fingerprints.ShouldShowSeekBarFingerprint
 
 @Patch
 @Name("Show seekbar")
@@ -21,11 +20,21 @@ import org.jf.dexlib2.iface.instruction.formats.Instruction22t
 class ShowSeekbarPatch : BytecodePatch(
     listOf(
         SetSeekBarShowTypeFingerprint,
+        ShouldShowSeekBarFingerprint,
     )
 ) {
     override fun execute(context: BytecodeContext): PatchResult {
+        ShouldShowSeekBarFingerprint.result?.mutableMethod?.apply {
+            addInstructions(
+                0,
+                """
+                    const/4 v0, 0x1
+                    return v0
+                """
+            )
+        }
         SetSeekBarShowTypeFingerprint.result?.mutableMethod?.apply {
-            val typeRegister = getInstruction<Instruction22t>(1).registerB
+            val typeRegister = implementation!!.registerCount - 1
 
             addInstructions(
                 0,

